tests/components/accuweather/test_config_flow.py

```diff
@@ -0,0 +1,158 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={
+            CONF_NAME: "abcd",
+            CONF_API_KEY: "foo",
+            CONF_LATITUDE: 55.55,
+            CONF_LONGITUDE: 122.12,
+        },
+    )
+
+    assert result["errors"] == {CONF_API_KEY: "invalid_api_key"}
+
+
+async def test_invalid_api_key_2(hass):
+    """Test that errors are shown when API key is invalid."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        side_effect=InvalidApiKeyError("Invalid API key"),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["errors"] == {CONF_API_KEY: "invalid_api_key"}
+
+
+async def test_api_error(hass):
+    """Test API error."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        side_effect=ApiError("Invalid response from AccuWeather API"),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_requests_exceeded_error(hass):
+    """Test requests exceeded error."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        side_effect=RequestsExceededError(
+            "The allowed number of requests has been exceeded"
+        ),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["errors"] == {CONF_API_KEY: "requests_exceeded"}
+
+
+async def test_integration_already_exists(hass):
+    """Test we only allow a single config flow."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        return_value=json.loads(load_fixture("accuweather/location_data.json")),
+    ):
+        MockConfigEntry(
+            domain=DOMAIN, unique_id="123456", data=VALID_CONFIG,
+        ).add_to_hass(hass)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["type"] == "abort"
+        assert result["reason"] == "single_instance_allowed"
+
+
+async def test_create_entry(hass):
+    """Test that the user step works."""
+    with patch(
+        "accuweather.AccuWeather._async_get_data",
+        return_value=json.loads(load_fixture("accuweather/location_data.json")),
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=VALID_CONFIG,
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "abcd"
+        assert result["data"][CONF_NAME] == "abcd"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)